project/src/main/java/com/google/sps/servlets/SellInvestmentServlet.java

```diff
@@ -0,0 +1,124 @@
+package com.google.sps.servlets;
+
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.Date;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import javax.sql.DataSource;
+
+/**
+ * Receive data when an investment is sold and updated the investments and participants database.
+ */
+@WebServlet("/sellInvestment")
+public class SellInvestmentServlet extends HttpServlet {
+
+  private static final Logger LOGGER = Logger.getLogger(SellInvestmentServlet.class.getName());
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    long investmentID = -1, priceSold = -1;
+    Date today = new Date();
+    long userID = -1;
+    long compID = -1;
+
+    String frontEndInfo = getInformationFromFrontEnd(request);
+
+    /**
+     * This part of code need to received a json obj from the front end to work
+     *
+     try {
+     //get specific information from json object
+     JSONObject jsonObj = new JSONObject(frontEndInfo);
+     investmentID = jsonObj.getLong("investmentID");
+     priceSold = jsonObj.getLong("priceSold");
```

Comment:
Can you extract this from the backend? What if an "attacker" changes priceSold at the frontend by modifying json and sells it for 1 Billion :).

Reply:
Yes! I got confused with buying and thought I need to get users input from frontend.